src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -590,6 +629,71 @@ namespace shogun
 			    ->cholesky_solver(L, b, lower);
 		}
 
+		/**
+		 * Compute the LDLT cholesky decomposition \f$A = P^{T} L D L^{*} P\f$
+		 * or \f$A = P^{T} U^{*} D U P\f$
+		 * of a positive semidefinite or negative semidefinite Hermitan matrix
+		 *
+		 * @param A The matrix whose LDLT cholesky decomposition is to be
+		 *  computed
+		 * @param L The matrix that saves the triangular LDLT
+		 *  Cholesky factorization (default: lower)
+		 * @param d The vector that saves the diagonal of the diagonal matrix D
+		 * @param p The vector that saves the permutation matrix P as a
+		 * transposition sequence
+		 * @param lower Whether to use L as the upper or lower triangular
+		 *  Cholesky factorization (default:lower)
+		 */
+		template <typename T>
+		void ldlt_factor(
+		    const SGMatrix<T>& A, SGMatrix<T>& L, SGVector<T>& d,
+		    SGVector<index_t>& p, const bool lower = true)
+		{
+			REQUIRE(
```

Comment:
@lisitsyn we need some systematic and compact way to do these checks. The amount of lines of code for such error messages is crazy